pkg/envoy/resources.go

```diff
@@ -27,9 +37,92 @@ const (
 	NetworkPolicyHostsTypeURL = "type.googleapis.com/cilium.NetworkPolicyHosts"
 )
 
+// NPHDSCache is a cache of resources in the Network Policy Hosts Discovery
+// Service.
+type NPHDSCache struct {
+	*xds.Cache
+}
```

Comment:
Out of scope of this review, and only out of curiosity. Is there a reason we embedded the type rather than `type NPHDSCache *xds.Cache` This way NPHDSCache also gets some functions from `xds.Cache` I think, or perhaps we have some future plans to add more fields?

Reply:
No reason, I thought that they are functionally equivalent.

Reply:
I had to check to figure it out. It is unintuitive :/ `type B *A` hides the public functions of A, `type B {*A}` doesn't. (https://play.golang.org/p/ri6K5HzLRy4)
It doesn't matter much but it would mean that a `NPHDSCache` implements all interfaces that `*xds.Cache` does. What might be more problematic is that you might not be able to define a function on `NPHDSCache` that has the same name as `*xds.Cache` because go doesn't have method overloading. :/ today-I-remembered go things

Reply:
In this case we need this type to also be of type `*xds.Cache`. That’s intentional.